File: LeetCode/python3/96.py
class Solution:
    def numTrees(self, n: int) -> int:
        trees = [0 for _ in range(n + 1)]
        trees[0], trees[1] = 1, 1,
        for i in range(2, n + 1):
            for j in range(1, i + 1):
                trees[i] += trees[j - 1] * trees[i - j]
        return trees[n]

File: LeetCode/cpp/96.cpp
class Solution {
public:
    int numTrees(int n) {
        vector<int> trees(n + 1, 0);
        trees[0] = 1;
        trees[1] = 1;
        for (int i = 2; i <= n; ++i)
            for (int j = 1; j <= i; ++j)
                trees[i] += trees[j - 1] * trees[i - j];
        return trees[n];
    }
};